File: data structures/stack/cs/Stack/Stack.cs
using System;
using System.Collections.Generic;

namespace Stack {
    ///<summary>
    /// Stack implementation on top a list.
    ///</summary>
    public class Stack<T> {

        private List<T> _Stack { get; } = new List<T> ();

        public void Push (T item) => _Stack.Add (item);

        public T Pop () => _Stack[_Stack.Count - 1];

        public int Count { get => _Stack.Count; }

        public bool IsEmpty { get => _Stack.Count == 0; }
    }
}

File: Stack/java/Stack.java

import java.util.*; 

public class Stack{
	private	int top;
	private	int[] stack;
	public Stack(){
		top = 0;
		stack = new int[5];
	}

	public void push(int value){
		if(this.top == stack.length) {
			throw new StackOverflowError();
		}
		else{
			this.stack[top++]=value;
		}
	}
	
	public int pop(){
		if(top == 0) {
			throw new IllegalStateException();
		}
		else{
			return this.stack[top--];
		}
	}

	public int peek(){
		if(top == 0) {
			throw new IllegalStateException();
		}
		else{
			return this.stack[top-1];
		}
	}

	public Boolean isEmpty(){
		return top==0;
	}

	@Override
	public String toString(){
		int[] content = Arrays.copyOfRange(stack, 0, top);
		return Arrays.toString(content);
	}
}